dataRetriever/resolvers/transactionResolver.go

```diff
@@ -156,19 +159,19 @@ func (txRes *TxResolver) resolveTxRequestByHashArray(hashesBuff []byte, pid p2p.
 		txsBuffSlice = append(txsBuffSlice, tx)
 	}
 
-	buffsToSend, err := txRes.dataPacker.PackDataInChunks(txsBuffSlice, maxBuffToSendBulkTransactions)
-	if err != nil {
-		return err
+	buffsToSend, errPack := txRes.dataPacker.PackDataInChunks(txsBuffSlice, maxBuffToSendBulkTransactions)
+	if errPack != nil {
+		return errPack
 	}
 
 	for _, buff := range buffsToSend {
-		err = txRes.Send(buff, pid)
-		if err != nil {
-			return err
+		errSend := txRes.Send(buff, pid)
+		if errSend != nil {
+			return errSend
 		}
 	}
 
-	return nil
+	return err
```

Comment:
the same situation here. Maybe return also counter for how many you managed to send.

Reply:
done. Added TODO to remove duplicate code